mockito-core/src/main/java/org/mockito/internal/creation/bytebuddy/InlineDelegateByteBuddyMockMaker.java

```diff
@@ -575,6 +565,15 @@ public String nonMockableReason() {
                 if (EXCLUDES.contains(type)) {
                     return "Cannot mock wrapper types, String.class or Class.class";
                 }
+                if (Modifier.isAbstract(type.getModifiers())
+                        && TypeSupport.INSTANCE.isSealed(type)) {
+                    if (type.isEnum()) {
+                        return "Sealed abstract enums can't be mocked. Since Java 15 abstract enums are declared sealed, which prevents mocking. You can still return an existing enum literal from a stubbed method call.";
+                    } else {
+                        return "Sealed interfaces or abstract classes can't be mocked. Interfaces cannot be instantiated and cannot be subclassed for mocking purposes. Instead of mocking a sealed interface or an abstract class, a non-abstract class can be mocked and used to represent the interface.";
```

Comment:
Nit: use `join` to make it multiple lines and a bit more readable.

Reply:
If I didn't it's because this string will appear has single reason prefixed by `- `. Joining will land the next line on column 0.

Tbis can be worked around though.